sotodlib/mapmaking/tod_grouping.py

```diff
@@ -0,0 +1,192 @@
+# this class intends to produce a obslists dictionary that will be feed to the mapmaker. Every element of obslists will be mapped.
+import numpy as np
+from pixell import utils
+from scipy import ndimage
+import so3g.proj
+
+from .utilities import *
+
+def find_scan_periods(obs_info, ttol=60, atol=2*utils.degree, mindur=120):
+    """Given a scan db, return the set of contiguous scanning periods in the form
+    [:,{ctime_from,ctime_to}]."""
+    atol = atol/utils.degree
+    info = np.array([obs_info[a] for a in ["az_center", "el_center", "az_throw", "timestamp", "duration"]]).T
+    # Get rid of nan entries
+    bad  = np.any(~np.isfinite(info),1)
+    # get rid of too short tods, since those don't have reliable az bounds
+    bad |= info[:,-1] < mindur
+    info = info[~bad]
+    t1   = info[:,-2] # timestamp is start of tod, unlike in enki
+    info = info[np.argsort(t1)]
+    # Start, end
+    t1   = info[:,-2]
+    t2   = t1 + info[:,-1]
+    # Remove angle ambiguities
+    info[:,0] = utils.rewind(info[:,0], period=360)
+    # How to find jumps:
+    # 1. It's a jump if the scanning changes
+    # 2. It's also a jump if a the interval between tod-ends and tod-starts becomes too big
+    changes    = np.abs(info[1:,:3]-info[:-1,:3])
+    jumps      = np.any(changes > atol,1)
+    jumps      = np.concatenate([[0], jumps]) # from diff-inds to normal inds
+    # Time in the middle of each gap
+    gap_times = np.mean(find_period_gaps(np.array([t1,t2]).T, ttol=ttol),1)
+    gap_inds  = np.searchsorted(t1, gap_times)
+    jumps[gap_inds] = True
+    # raw:  aaaabbbbcccc
+    # diff: 00010001000
+    # 0pre: 000010001000
+    # cum:  000011112222
+    labels  = np.cumsum(jumps)
+    linds   = np.arange(np.max(labels)+1)
+    t1s     = ndimage.minimum(t1, labels, linds)
+    t2s     = ndimage.maximum(t2, labels, linds)
+    # Periods is [nperiod,{start,end}] in ctime. Start is the start of the first tod
+    # in the scanning period. End is the end of the last tod in the scanning period.
+    periods = np.array([t1s, t2s]).T
+    return periods
+
+def find_scan_periods_perobs(obs_info, mindur=120):
+    """Given a scan db, return the periods per obs, i.e. start and stop of each scan. This is a simplified version of find_scan_periods
```

Comment:
this function is not be called outside, so I think it doesn't need a docstring. A scan db is an obslist (the output of `obsdb.query(...)`)

Reply:
Ok, perhaps you can just change the language to be "a list of obsdb info from an obsdb.query()" instead of scan db as I don't think scan db is standard language across the package.